modules/common/react/lib/utils/useUniqueId.ts

```diff
@@ -7,8 +7,14 @@ import uuid from 'uuid/v4';
  */
 export const useUniqueId = (id?: string) => {
   // https://codesandbox.io/s/react-functional-component-ids-p2ndq
-  const [generatedId] = React.useState(() => uuid().replace(/^[0-9]*/gi, ''));
-  return id || generatedId;
+  const [effectiveId] = React.useState(() => {
+    if (typeof id !== 'undefined') {
+      return id;
+    }
+    return uuid().replace(/^[0-9]*/gi, '');
+  });
+
+  return effectiveId;
```

Comment:
I'm curious about this change. It looks like the only behavior difference is not calling `uuid` unless the `id` is `undefined` vs always calling `uuid` and determining the truthiness of `id`.

Reply:
Yea, that's the gist of it. It seemed wasteful to compute a uuid even when we want to just pass the id through if it's defined.

Reply:
Yes, you're right about that. Let me revert this change, I feel like it's not really related to this PR.